### PR TITLE
fix test harness to handle crazy version numbers

### PR DESCRIPTION
**Pull request by joshudson**
I kept local style in this function rather than documented style guide style.

Fixes: #38008

    eng/common/dotnet-install.sh

changes permissions to 0755 every build; might as well commit it now.

### CODE REVIEW

File: src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.Unix.cs [hunk 1]
```diff
@@ -157,12 +157,30 @@ private static Version GetICUVersion()
...
+            if (index < versionString.Length)
+                versionString = versionString.Substring(0, index);
+            try
+            {
+               if (versionString.IndexOf('.') != -1)
+                   return new Version(versionString);
+
+               // minor version is required by Version
+               // let's default it to 0
+               return new Version(int.Parse(versionString), 0);
```
Comment (danmosemsft) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.Unix.cs]: TryParse instead of throw/catch?

**Comment by danmosemsft** [comment]
This problem seems to have been solved several times.
https://github.com/dotnet/corefx/blob/b86783ef9525f22b0576278939264897464f9bbd/src/Common/src/CoreLib/System/Environment.Unix.cs#L313
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs#L196

I wonder if there is a way to avoid a 3rd version.

**Comment by joshudson** [comment]
If you think you can get that ParseVersion from FileVersionInfo.Unix past the API review team as a public API surface, go for it. I just need to get this broken test runner out of my way. I'd care if it was the shipping code rather than the test harness code. I'll gladly use TryParse though.

**Comment by danmosemsft** [comment]
If the procedure in Environment.Unix.cs would work, you could extract it into eg corefx/src/Common/src/CoreLib/System/VersionUtilities.cs and then pull that file into this project (like is already done with eg PasteArguments.cs)

@jkotas seem reasonable?

**Comment by joshudson** [comment]
Yeah that one's actually a little better even. Unix world doesn't constrain themselves to 16 bit version number components.

This comes up a lot in my code too. I've got a bunch of slightly different little version number parsers. All of a sudden fixing this right with a public interface after 3.0 release sounds attractive.

Apologies for the dithering; it's late and I need to go to bed.

**Comment by danmosemsft** [comment]
> fixing this right with a public interface after 3.0 release sounds attractive.

Worth considering - for now my suggestion was just code reuse.

**Comment by jkotas** [comment]
@ViktorHofer This change is dotnet/arcade already. The problem is that the script that copies it around does not preserve Unix file permissions, tracked by https://github.com/dotnet/arcade/issues/2862 . This manual change is working around it.

**Comment by jhudsoncedaron** [comment]
The build failures look more like build timeouts. Here's the linux x64 one:

.packages/microsoft.dotnet.helix.sdk/2.0.0-beta.19278.1/tool/Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(67,5): error : This task can only be used on completed jobs. There are 30 of 212 unfinished work items.
eng/common/tools.sh(426,): error : Build failed (exit code '1').
Bash exited with code '1'.

**Comment by wfurt** [comment]
Did we converge on approach? This blocks 
Stand up Debian 10 ("Buster") Helix queues https://github.com/dotnet/core-eng/issues/6166

**Comment by jhudsoncedaron** [comment]
I'm gonna put an API change request in after 3.0; but now that you're actually trying to use Buster you should consider merging this in the meantime.

**Comment by danmosemsft** [comment]
https://github.com/dotnet/corefx/issues/38443#issuecomment-503183394 needs to be fixed in a similar place. eg
https://github.com/dotnet/corefx/compare/master...danmosemsft:distro.version

**Comment by maryamariyan** [comment]
@jhudsoncedaron master branch is now open for 5.0. 
This PR has been inactive for a while. What are the next steps forward for this?

If this PR is awaiting other changes, it would be best to close this and reopen when this is ready for review.

**Comment by maryamariyan** [comment]
I also noticed the issue you link for this PR is to fix is already closed (#39380

**Comment by joshudson** [comment]
This PR is obsolete.

**Pull request closed without merging**